lib/Rfc1035StubResolver.php

```diff
@@ -232,9 +257,22 @@ public function query(string $name, int $type): Promise
         }
 
         $promise = call(function () use ($name, $type) {
-            if (!$this->config) {
+            if ($this->configStatus === self::CONFIG_NOT_LOADED) {
                 yield $this->reloadConfig();
             }
+            if ($this->configStatus === self::CONFIG_FAILED) {
+                if ($type !== Record::A) {
+                    throw new NoRecordException("No records returned for '{$name}' (".Record::getName($type).")");
```

Comment:
Should say: `Query failed, because loading the system's DNS configuration failed and querying records other than A records isn't supported in blocking fallback mode.`